framework/data_types/ndarray.h

```diff
@@ -446,94 +242,104 @@ class NDArray
   template <typename D>
   void resize(const std::initializer_list<D>& dims)
   {
-    static_assert(std::is_integral<D>::value,
-                  "NDArray dims argument must have vector of integral types.");
-
-    dimensions_.clear();
-    strides_.clear();
-    delete[] base_;
-    base_ = nullptr;
-
-    NDArray<T>(dims).swap(*this);
+    set_dimensions(dims);
+    if (size_ != std::accumulate(dims.begin(), dims.end(), 1, std::multiplies<size_t>()))
+      initialize();
   }
 
   /**
-   * Accesses the specified element.
+   * Accesses the specified element for an array with N dimensions.
    *
-   * \param args The indices of the desired element.
+   * \param args Indices for each dimension.
    * \return Read/write reference to the element.
    */
   template <typename... Args>
-  T& operator()(Args... args) noexcept
+  inline __attribute__((always_inline)) constexpr T& operator()(Args... args) noexcept
```

Comment:
This is some notation I haven't seen before... Could you explain to a layman what this means? In particular, I've never seen the `__attribute__((always_inline))` and don't fully understand why this should be marked `constexpr`

Reply:
`inline` by itself is just a hint to the compiler. I'm using a gcc/clang specific compiler directive (`__attribute__((always_inline))`) to tell the compiler I really want you to inline this. It's a `constexpr` to try to get as much performance out of it as possible. It also calls a `constexpr` method.